sdk/nodejs/cmd/run/index.ts

```diff
@@ -111,8 +111,9 @@ function main(args: string[]): void {
     v8Hooks.isInitializedAsync().then(() => {
         require("./run").run(argv, () => {
             programRunning = true;
+        }).then(() =>  {
+            programRunning = false;
         });
-        programRunning = false;
```

Comment:
@pgavlin I'd like to discuss this.  It's not even clear to me that we need to set programRunning to false in the first place.  but if that's important here, then i wanted to added it back in.

We should talk in person, however the problem with the existing code is that it called into "run", hooks up the secondary error-handlers, then calls the first lambda (setting `programRunning` to true). However, `run` then fires off `runtime.runInPulumiStack` which is *async*.  So `run` immediately returns, and we set `programRunning` back to `false`.  This means that the error-handler you hooked up always says "i should print out this error".

Reply:
Ah, good catch.

Reply:
I would prefer that we continue setting `programRunning` back to `false`. IIUC, not doing so means that we might miss stack traces/messages/etc for unhandled errors that occur after the program's handler has been unregistered (assuming that is a thing that happens).

Reply:
That is totally fine with me.